src/core/mac/link_raw.cpp

```diff
@@ -51,17 +51,27 @@ namespace Mac {
 
 LinkRaw::LinkRaw(Instance &aInstance)
     : InstanceLocator(aInstance)
-    , mReceiveChannel(OPENTHREAD_CONFIG_DEFAULT_CHANNEL)
-    , mPanId(kPanIdBroadcast)
-    , mReceiveDoneCallback(nullptr)
-    , mTransmitDoneCallback(nullptr)
-    , mEnergyScanDoneCallback(nullptr)
 #if OPENTHREAD_RADIO
     , mSubMac(aInstance)
 #elif OPENTHREAD_CONFIG_LINK_RAW_ENABLE
     , mSubMac(aInstance.Get<SubMac>())
 #endif
 {
+    Init();
+}
+
+void LinkRaw::Init(void)
+{
+    mEnergyScanDoneCallback = nullptr;
+    mTransmitDoneCallback   = nullptr;
+    mReceiveDoneCallback    = nullptr;
```

Comment:
Is it necessary to reset these pointers?

Reply:
`mEnergyScanDoneCallback` and `mTransmitDoneCallback` are temporary callbacks, they are set every time when scans or transmits.

`mReceiveDoneCallback` is used as the indicator of whether the raw link layer is enabled, it's necessary to reset to disabled state.